### PR TITLE
BZ1980340: [DOC-OCPRHV] Add resize to ovirt CSI driver

### PR DESCRIPTION
**Pull request by stoobie**
enterprise_4.9

For versions 4.9+

https://bugzilla.redhat.com/show_bug.cgi?id=1980340

Direct link to doc preview: 
deploy-preview-34475--osdocs.netlify.app/?utm_source=github&utm_campaign=bot_dp

For each item in bug comment 2:

1. CSI drivers supported by OpenShift container platform Table 1:

https://deploy-preview-34475--osdocs.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi.html#csi-drivers-supported_persistent-storage-csi

2. and 4. oVirt CSI driver storage class (code example):
https://deploy-preview-34475--osdocs.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-ovirt.html#ovirt-csi-driver-storage-class_persistent-storage-csi-ovirt

3. Creating a persistent volume on RHV > pvc-ovirt.yaml:
https://deploy-preview-34475--osdocs.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-ovirt.html#persistent-storage-rhv_persistent-storage-csi-ovirt

5. Add a section for RHV/oVirt just as there are specifics for other providers. Use code example from previous item.
https://deploy-preview-34475--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/storage-configuration.html#ovirt-csi-driver-storage-class_post-install-storage-configuration

### CODE REVIEW

**Comment by stoobie** [comment]
@Gal-Zaidman Please review.

**Comment by Gal-Zaidman** [comment]
- Not sure where it is in the code, but the title on[1] needs to be renamed to "Red Hat Virtualization (RHV) object definition" to align to the rest of the platforms.
- The storage is missing one more space

[1] https://deploy-preview-34475--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/storage-configuration.html#ovirt-csi-driver-storage-class_post-install-storage-configuration

File: modules/persistent-storage-rhv.adoc [hunk 1]
```diff
@@ -45,7 +45,7 @@ spec:
     - ReadWriteOnce
   resources:
     requests:
-    storage: <volume size>  <2>
+     storage: <volume size>  <2>
```
Comment (Gal-Zaidman) [inline modules/persistent-storage-rhv.adoc:48]: please add another space (2 spaces required)

**Comment by stoobie** [comment]
>     * Not sure where it is in the code, but the title on[1] needs to be renamed to "Red Hat Virtualization (RHV) object definition" to align to the rest of the platforms.

Is the rest of the topic OK as-is?

File: modules/persistent-storage-rhv.adoc (hunk 1 above)
Comment (stoobie) [reply modules/persistent-storage-rhv.adoc]: @Gal-Zaidman  Fixed.

**Comment by Gal-Zaidman** [comment]
Things look OK to me I think you fixed all my comments

**Review by Gal-Zaidman — APPROVED** [review]

**Comment by stoobie** [comment]
@Gal-Zaidman : @mgold1234 thinks it would be helpful to add explanations of these two items:

`reclaimPolicy: Delete`
`volumeBindingMode: Immediate`

What do you think?

File: modules/ovirt-csi-driver-storage-class.adoc [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
 apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   name: <storage-class-name>  <1>
   annotations:
-    storageclass.kubernetes.io/is-default-class: "false"  <2>
+    storageclass.kubernetes.io/is-default-class: "<boolean>"  <2>
 provisioner: csi.ovirt.org
+allowVolumeExpansion: <boolean> <3>
+reclaimPolicy: Delete
```
Comment (mgold1234) [inline modules/ovirt-csi-driver-storage-class.adoc:30]: reclaimPolicy - Dynamically provisioned PersistentVolumes of this storage class are created with this reclaimPolicy. Defaults to Delete.

File: modules/ovirt-csi-driver-storage-class.adoc [hunk 3]
```diff
@@ -1,30 +1,42 @@
...
 ----
 apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   name: <storage-class-name>  <1>
   annotations:
-    storageclass.kubernetes.io/is-default-class: "false"  <2>
+    storageclass.kubernetes.io/is-default-class: "<boolean>"  <2>
 provisioner: csi.ovirt.org
+allowVolumeExpansion: <boolean> <3>
```
Comment (mgold1234) [inline modules/ovirt-csi-driver-storage-class.adoc:29]: allowVolumeExpansion -  AllowVolumeExpansion shows whether the storage class allow volume expand

Comment (mgold1234) [reply modules/ovirt-csi-driver-storage-class.adoc:29]: I think it is good to add some explanation

File: modules/ovirt-csi-driver-storage-class.adoc [hunk 4]
```diff
@@ -1,30 +1,42 @@
...
 ----
 <1> Name of the storage class.
 <2> Set to `false` if the storage class is the default storage class in the cluster. If set to `true`, the existing default storage class must be edited and set to `false`.
-<3> {rh-virtualization} storage domain name to use.
-<4> Disk must be thin provisioned.
-<5> File system type to be created.
+<3> `true` enables dynamic volume expansion, `false` prevents it. `true` is recommended.
+<4> The {rh-virtualization} storage domain name to use.
+<5> If `true`, the disk is thin provisioned. If `false`, the disk is preallocated. Thin provisioning is recommended.
+<6> File system type to be created. Possible values: `ext4` or `xfs`.
```
Comment (mgold1234) [inline modules/ovirt-csi-driver-storage-class.adoc:42]: this parameter is optional, default is ext4 (user don't have to add this row to yaml)

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 2 above)
Comment (stoobie) [reply modules/ovirt-csi-driver-storage-class.adoc:30]: @Gal-Zaidman @mgold1234 What are the other values? What does `Delete` do?

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 3 above)
Comment (stoobie) [reply modules/ovirt-csi-driver-storage-class.adoc:29]: @mgold1234 Can you be more specific? What do you want to be explained here that isn't already?

Comment (stoobie) [reply modules/ovirt-csi-driver-storage-class.adoc:29]: > allowVolumeExpansion - AllowVolumeExpansion shows whether the storage class allow volume expand

@mgold1234  Doesn't line 39 below already explain this?

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 2 above)
Comment (mgold1234) [reply modules/ovirt-csi-driver-storage-class.adoc:30]: @stoobie , I think you can add explanation to reclaimPolicy: Delete
volumeBindingMode: Immediate. VolumeBindingMode -  indicates how PersistentVolumeClaims should be provisioned and bound. When unset, VolumeBindingImmediate is used. This field is only honored by servers that enable the VolumeScheduling feature.

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 3 above)
Comment (mgold1234) [reply modules/ovirt-csi-driver-storage-class.adoc:29]: yes, I agree line 39 is enough

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 2 above)
Comment (stoobie) [reply modules/ovirt-csi-driver-storage-class.adoc]: Done.

File: modules/ovirt-csi-driver-storage-class.adoc [hunk 5]
```diff
@@ -1,30 +1,44 @@
...
 
 To create additional storage classes for different configurations, create and save a file with the `StorageClass` object described by the following sample YAML:
 
+.ovirt-storageclass.yaml
 [source,yaml]
 ----
 apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   name: <storage-class-name>  <1>
```
Comment (bobfuru) [inline modules/ovirt-csi-driver-storage-class.adoc:25]: Could you please make this `storage_class_name` (use underscores) to be in line with our convention? Thanks!

File: modules/ovirt-csi-driver-storage-class.adoc [hunk 6]
```diff
@@ -1,30 +1,44 @@
...
 <2> Set to `false` if the storage class is the default storage class in the cluster. If set to `true`, the existing default storage class must be edited and set to `false`.
-<3> {rh-virtualization} storage domain name to use.
-<4> Disk must be thin provisioned.
-<5> File system type to be created.
+<3> `true` enables dynamic volume expansion, `false` prevents it. `true` is recommended.
+<4> Dynamically provisioned persistent volumes of this storage class are created with this reclaim policy. This default policy is `Delete`.
+<5> Indicates how to provision and bind `PersistentVolumeClaims`. When not set, `VolumeBindingImmediate` is used. This field is only applied by servers that enable the `VolumeScheduling` feature.
+<6> The {rh-virtualization} storage domain name to use.
+<7> If `true`, the disk is thin provisioned. If `false`, the disk is preallocated. Thin provisioning is recommended.
+<8> (Optional) File system type to be created. Possible values: `ext4` (default) or `xfs`.
```
Comment (bobfuru) [inline modules/ovirt-csi-driver-storage-class.adoc:44]: s/(Optional) File/Optional: File
This is how we generally introduce an optional step or value (with a colon).

**Review by bobfuru — COMMENTED** [review]
A couple of minor requests and otherwise this LGTM! Also, not a part of this PR but I noticed a typo in the module "Creating a persistent volume on RHV" (`persistent-storage-rhv.adoc`), the first line of the procedure: "If you are using the we console...." Would you mind fixing this as well? 🙇

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 5 above)
Comment (stoobie) [reply modules/ovirt-csi-driver-storage-class.adoc:25]: Fixed. Thanks for pointing this out.

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 6 above)
Comment (stoobie) [reply modules/ovirt-csi-driver-storage-class.adoc:44]: Fixed. The IBM SG actually says the same thing you did.
However, I found two other instances of this in files that I don't own. I went ahead and made the changes there too, even though those files are not connected to this PR. I can undo the changes if you like.

**Comment by stoobie** [comment]
@bobfuru I made the changes you suggested. Please approve and set this in the merge queue, or let me know what else should change.

File: modules/ovirt-csi-driver-storage-class.adoc (hunk 6 above)
Comment (bobfuru) [reply modules/ovirt-csi-driver-storage-class.adoc]: That's fantastic, thanks for taking care of all these.

**Comment by bobfuru** [comment]
LGTM - will merge/CP to 4.9

**Pull request merged**